onnx/test/optimizer_test.py

```diff
@@ -93,11 +93,29 @@ def test_fuse_transpose_into_gemm(self):
              helper.make_tensor_value_info("Y", TensorProto.FLOAT, (5, 2)),
              helper.make_tensor_value_info("C", TensorProto.FLOAT, (3, 5))],
             [helper.make_tensor_value_info("Z", TensorProto.FLOAT, (3, 5))])
-        optimized_model = self._optimized(graph, "fuse_transpose_into_gemm")
+        optimized_model = self._optimized(graph, ["fuse_transpose_into_gemm"])
 
         assert len(list(optimized_model.graph.node)) == 1
         assert optimized_model.graph.node[0].op_type == "Gemm"
 
+    def test_preserve_value_info(self):
+        trans1 = helper.make_node("Transpose", ["X"], ["Y"], perm=[1,0,2])
+        trans2 = helper.make_node("Transpose", ["Y"], ["Z"], perm=[2,0,1])
+        trans3 = helper.make_node("Transpose", ["Z"], ["A"], perm=[2,0,1])
+        graph = helper.make_graph(
+            [trans1, trans2, trans3],
+            "test",
+            [helper.make_tensor_value_info("X", TensorProto.FLOAT, (2, 3, 4))],
+            [helper.make_tensor_value_info("A", TensorProto.FLOAT, (4, 3, 2))])
+        graph.value_info.extend(
+            [helper.make_tensor_value_info("Y", TensorProto.FLOAT, (3, 2, 4))]
+        )
+
+        optimized_model = self._optimized(graph, [])
+
+        assert len(optimized_model.graph.value_info) == 1
```

Comment:
You may also want to check value_info's name, type and shape.

Reply:
sure ++